ergotree-ir/src/mir/value.rs

```diff
@@ -69,6 +69,28 @@ impl CollKind {
         }
     }
 
+    /// Build a collection from items where each is a collection as well, storing them as Rust types values when neccessary
+    pub fn from_vec_vec(
```

Comment:
I wonder if `from_vec_vec`(any level of "nestedness" really) could be implemented in `from_vec` with a recursive call. `elem_tpe` will be `SColl(...)`, etc.

Reply:
I don't see any obvious way. Flattening step `Vec<Vec<A>>  -> Vec<A>` must come from somewhere. I'll think about it. I don't like proposed implementation anyway

Reply:
I appreciate your efforts! Let me know if you want me to merge this PR as is.

Reply:
Looks like it's not possible to do better with iterators. They don't support chaining of several failible computations. Something like `.map(f:A->Result<B,Err>).map(g:B->Result<C,Err>)`. It's not possible to push error into iterator since `next` returns only `Option` and can't signal error easily.

Only thing I managed to improve is to use partial type hints to guide type inference.